app/models/specific_tube_rack_creation.rb

```diff
@@ -0,0 +1,436 @@
+# frozen_string_literal: true
+# Allows for the creation of multiple tube racks and their tubes.
+# rubocop:disable Metrics/ClassLength
+class SpecificTubeRackCreation < AssetCreation
+  # Allows a many to many relationship between SpecificTubeRackCreations and child Tube racks.
+  class ChildTubeRack < ApplicationRecord
+    self.table_name = ('specific_tube_rack_creation_children')
+    belongs_to :specific_tube_rack_creation
+    belongs_to :tube_rack
+  end
+
+  # Allows a many to many relationship between SpecificTubeRackCreations and Purposes.
+  class ChildPurpose < ApplicationRecord
+    self.table_name = 'specific_tube_rack_creation_purposes'
+    belongs_to :specific_tube_rack_creation
+    belongs_to :tube_rack_purpose, class_name: 'Purpose'
+  end
+
+  has_many :creation_child_purposes, class_name: 'SpecificTubeRackCreation::ChildPurpose', dependent: :destroy
+  has_many :child_purposes, through: :creation_child_purposes, source: :tube_rack_purpose
+
+  has_many :child_tube_racks, class_name: 'SpecificTubeRackCreation::ChildTubeRack', dependent: :destroy
+  has_many :children, through: :child_tube_racks, source: :tube_rack
+
+  validates :tube_rack_attributes, presence: true
+
+  has_many :parent_associations,
+           foreign_key: 'asset_creation_id',
+           class_name: 'AssetCreation::ParentAssociation',
+           inverse_of: 'asset_creation',
+           dependent: :destroy
+
+  # NB. assumption that parent is a plate
```

Comment:
I think making this assumption is OK if it makes things easier. However, just wondering - did you try having it as 'Labware' instead of Plate?

Reply:
Use case for this endpoint is plate to tube rack creation. Didn't want to get into the complexities of going from a tube to a tuberack (how would that work?) or tuberack to tuberack. Can discuss.

Reply:
Think those would be different endpoints.